client/landing/stepper/declarative-flow/internals/steps-repository/domain-transfer-domains/domain-code-pair.tsx

```diff
@@ -60,7 +62,11 @@ const DomainPrice = ( { rawPrice, saleCost, currencyCode = 'USD' }: DomainPriceP
 	const { __ } = useI18n();
 
 	if ( ! rawPrice ) {
-		return <div className="domains__domain-price-number disabled">0</div>;
+		return (
+			<div className="domains__domain-price-number disabled">
```

Comment:
Testing locally, I can confirm that I'm getting more correct price formatting.
<img width="109" alt="Screenshot 2023-07-19 at 2 59 31 PM" src="https://github.com/Automattic/wp-calypso/assets/1126811/7998f68f-af8e-4de7-914f-6b99abacce3a">

Reply:
Currency gets geolocated for logged out users regardless of locale. "33 US$" is the kind of thing you see when you have a french locale but currency set to USD.

Logged in, you can change your user's currency via SA, that's usually how I test different currencies, to test the currency detection itself you need a vpn.